latch_cli/utils.py

```diff
@@ -115,27 +115,38 @@ def with_si_suffix(num, suffix="B", styled=False):
 def hash_directory(dir_path: Path) -> str:
     m = hashlib.new("sha256")
     m.update(current_workspace().encode("utf-8"))
-    for containing_path, dirnames, fnames in os.walk(dir_path):
+    exclude = None
+    ignore_file = dir_path / ".dockerignore"
+    if os.path.exists(ignore_file):
+        with open(ignore_file) as f:
```

Comment:
```py
exists: List[str] = []
```

then, remove the `exists` check and handle exceptions instead (there are other exceptions e.g. file access that you want to handle)

```py
for l in ignore_file.open("r"):
    l = l.strip()

		if l == "":
				continue
		if l[0] == "#":
        continue

    exists.append(l)
```